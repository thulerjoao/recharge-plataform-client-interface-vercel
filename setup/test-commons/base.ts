import type {Config} from '@jest/types'
import path from 'path'

//Sync object
const config: Config.InitialOptions = {
    testEnvironment: 'jsdom',
    verbose: true,
    setupFilesAfterEnv: [
        path.resolve(__dirname, 'src', 'setupTests.js')
    ],
    transform: {
        '\\.[jt]sx?$': ['babel-jest', {
            configFile: path.resolve(__dirname, '.babelrc')
        }],
    },
    moduleNameMapper: {
      '^axios$': require.resolve('axios'),
    },
    testMatch: [
      '**/__tests__/**/*.[jt]s?(x)',
      '**/?(*.)+(spec|test).[tj]s?(x)',
    ],
}
export default config

