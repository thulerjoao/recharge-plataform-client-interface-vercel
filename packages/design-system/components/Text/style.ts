import styled from "styled-components";
import { Theme } from "../../theme/theme";

interface TextProps {
  margin?: string;
  color?: string;
  threeDotsLimit?: boolean;
  fontSize: string;
  fontWeight: number;
  align?: string;
  nowrap: boolean;
  pointer: boolean;
  underline: boolean;
}

export const TextContainer = styled.p<TextProps>`
  ${({ margin }) => margin && `margin: ${margin};`};
  text-align: ${({ align }) => (!align ? "start" : align)};
  color: ${({ color }) => (color ? color : Theme.colors.mainlight)};
  font-size: ${({ fontSize }) => fontSize};
  font-weight: ${({ fontWeight }) => fontWeight};
  ${({ threeDotsLimit }) => threeDotsLimit && `white-space: nowrap;`};
  ${({ threeDotsLimit }) => threeDotsLimit && `overflow: hidden;`};
  ${({ threeDotsLimit }) => threeDotsLimit && `text-overflow: ellipsis;`};
  width: 100%;
  height: auto;
  white-space: ${({ nowrap }) => nowrap && "nowrap"};
  cursor: ${({ pointer }) => pointer && "pointer"};
  text-decoration: ${({ underline }) => underline && "underline"};
`;
