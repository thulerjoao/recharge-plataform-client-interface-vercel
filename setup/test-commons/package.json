{
    "name": "@4miga/test-commons",
    "version": "1.0.0",
    "private": true,
    "main": "index.js",
    "scripts": {
      "lint": "echo 'Not necessary'",
      "test": "echo 'Not necessary'"
    },
    "files":[
      "base.ts"
    ],
    "devDependencies": {
        "jest":"27.0.0",
        "ts-node":"10.7.0",
        "babel-jest":"28.0.3",
        "@babel/core":"7.17.10",
        "@babel/preset-env":"7.17.10",
        "@babel/preset-react":"7.16.7",
        "@babel/preset-typescript":"7.16.7",
        "@testing-library/react": "14.2.1",
        "jest-styled-components" : "7.2.0"
    }
  }
