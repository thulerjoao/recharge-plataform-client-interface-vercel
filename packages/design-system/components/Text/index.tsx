"use client";

import React, { useMemo } from "react";
import { StyleSheetManager } from "styled-components";
import { fontTypes } from "../../theme/modules/font";
import { TextContainer } from "./style";

interface TextProps extends React.HTMLAttributes<HTMLSpanElement> {
  align?: "center" | "start" | "end" | "justify";
  tag?: "h1" | "h2" | "h3" | "h4" | "h5" | "h6" | "p" | "span" | "u";
  fontType: fontTypes;
  color: string;
  margin?: string;
  nowrap?: boolean;
  pointer?: boolean;
  underline?: boolean;
  children: React.ReactNode;
}

const Text = ({
  margin,
  tag,
  align,
  fontType,
  color,
  nowrap,
  pointer,
  underline,
  children,
  ...props
}: TextProps) => {
  const fontSize = useMemo(() => {
    switch (fontType) {
      case "TINY":
      case "TINY_MEDIUM":
      case "TINY_SEMI_BOLD":
        return "12px";
      case "SMALL":
      case "SMALL_MEDIUM":
      case "SMALL_SEMI_BOLD":
        return "14px";
      case "REGULAR":
      case "REGULAR_MEDIUM":
      case "REGULAR_SEMI_BOLD":
        return "16px";
      case "LARGE":
      case "LARGE_MEDIUM":
      case "LARGE_SEMI_BOLD":
        return "20px";
      case "BIG":
      case "BIG_MEDIUM":
      case "BIG_SEMI_BOLD":
        return "24px";
      default:
        return "16px";
    }
  }, [fontType]);

  const fontWeight = useMemo(() => {
    switch (fontType) {
      case "TINY":
      case "SMALL":
      case "REGULAR":
      case "LARGE":
      case "BIG":
        return 400;
      case "TINY_MEDIUM":
      case "SMALL_MEDIUM":
      case "REGULAR_MEDIUM":
      case "LARGE_MEDIUM":
      case "BIG_MEDIUM":
        return 500;
      case "TINY_SEMI_BOLD":
      case "SMALL_SEMI_BOLD":
      case "REGULAR_SEMI_BOLD":
      case "LARGE_SEMI_BOLD":
      case "BIG_SEMI_BOLD":
        return 600;
      default:
        return 400;
    }
  }, [fontType]);

  return (
    <StyleSheetManager
      shouldForwardProp={(prop) =>
        prop !== "align" &&
        prop !== "margin" &&
        prop !== "fontSype" &&
        prop !== "fontWeight" &&
        prop !== "color" &&
        prop !== "nowrap" &&
        prop !== "pointer" &&
        prop !== "underline" &&
        prop !== "threeDotsLimit"
      }
    >
      <TextContainer
        margin={margin}
        align={align}
        as={tag}
        fontSize={fontSize}
        fontWeight={fontWeight}
        color={color}
        nowrap={nowrap}
        pointer={pointer}
        underline={underline}
        {...props}
      >
        {children}
      </TextContainer>
    </StyleSheetManager>
  );
};

export default Text;
