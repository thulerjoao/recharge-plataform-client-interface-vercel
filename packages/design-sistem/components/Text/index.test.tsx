import Text from "./index";
import { customRender } from "@4miga/test-commons/react-testing-library";

const render = customRender();

describe("<Text />", () => {
  it("render h1 tag", () => {
    const { container } = render(<Text tag="h2">Sample</Text>);
    expect(container).toMatchSnapshot();
  });
});
