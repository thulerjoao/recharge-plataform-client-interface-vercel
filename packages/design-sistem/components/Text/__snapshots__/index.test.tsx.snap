// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> render h1 tag 1`] = `
<div>
  <h2
    class=""
  >
    Sample
  </h2>
</div>
`;
